jwst/associations/lib/diff.py

```diff
@@ -207,9 +207,13 @@ def compare_asn_lists(left_asns, right_asns):
     # Ensure that the product name lists are the same.
     name_diff = left_product_names ^ right_product_names
     if name_diff:
+        left_not_right = left_product_names - right_product_names
+        right_not_left = right_product_names - left_product_names
```

Comment:
No need for `name_diff` at all then? Would this work?
```suggestion
    left_not_right = left_product_names - right_product_names
    right_not_left = right_product_names - left_product_names
    if left_not_right or right_not_left:
```